mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVaultTest.java

```diff
@@ -19,110 +19,112 @@
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.ByteArrayInputStream;
 import java.time.Instant;
-import java.time.ZoneId;
+import java.time.ZonedDateTime;
 
+import org.apache.james.blob.api.BucketName;
 import org.apache.james.blob.api.HashBlobId;
 import org.apache.james.blob.memory.MemoryBlobStore;
+import org.apache.james.utils.UpdatableTickingClock;
 import org.apache.james.vault.DeletedMessageVault;
 import org.apache.james.vault.DeletedMessageVaultContract;
+import org.apache.james.vault.RetentionConfiguration;
 import org.apache.james.vault.memory.metadata.MemoryDeletedMessageMetadataVault;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import reactor.core.publisher.Mono;
 
 class BlobStoreDeletedMessageVaultTest implements DeletedMessageVaultContract {
-    private static final Instant NOW = Instant.parse("2007-12-03T10:15:30.00Z");
-    private static final Clock CLOCK = Clock.fixed(NOW, ZoneId.of("UTC"));
 
     private BlobStoreDeletedMessageVault messageVault;
+    private UpdatableTickingClock clock;
 
     @BeforeEach
     void setUp() {
+        clock = new UpdatableTickingClock(NOW.toInstant());
         messageVault = new BlobStoreDeletedMessageVault(new MemoryDeletedMessageMetadataVault(),
             new MemoryBlobStore(new HashBlobId.Factory()),
-            new BucketNameGenerator(CLOCK));
+            new BucketNameGenerator(clock), clock, RetentionConfiguration.DEFAULT);
     }
 
     @Override
     public DeletedMessageVault getVault() {
         return messageVault;
     }
 
-
-    @Disabled("Will be implemented later")
-    public void deleteShouldThrowOnNullMessageId() {
-
-    }
-
-    @Disabled("Will be implemented later")
-    public void deleteShouldThrowOnNullUser() {
-
-    }
-
-    @Disabled("Will be implemented in JAMES-2811")
     @Override
-    public void deleteExpiredMessagesTaskShouldCompleteWhenNoMail() {
-
+    public UpdatableTickingClock getClock() {
+        return clock;
     }
 
-    @Disabled("Will be implemented in JAMES-2811")
-    @Override
-    public void deleteExpiredMessagesTaskShouldCompleteWhenAllMailsDeleted() {
+    @Test
+    void retentionQualifiedBucketsShouldReturnOnlyBucketsFullyBeforeBeginningOfRetentionPeriod() {
+        clock.setInstant(Instant.parse("2007-12-03T10:15:30.00Z"));
+        Mono.from(getVault().append(USER, OLD_DELETED_MESSAGE, new ByteArrayInputStream(CONTENT))).block();
+        clock.setInstant(Instant.parse("2008-01-03T10:15:30.00Z"));
+        Mono.from(getVault().append(USER, DELETED_MESSAGE_2, new ByteArrayInputStream(CONTENT))).block();
 
+        ZonedDateTime beginningOfRetention = ZonedDateTime.parse("2008-01-30T10:15:30.00Z");
+        assertThat(messageVault.retentionQualifiedBuckets(beginningOfRetention).toStream())
+            .containsOnly(BucketName.of("deleted-messages-2007-12-01"));
```

Comment:
This bucket also contains mail from `2007/12/25` that are not yet 1 year old and that you would delete. CF earlier comments.